rero/rero-ils-ui
# circulation: allow checkout with fixed due date

Adds a 'setting' button in the patron circulation component. This button
allows the staff member to choose some options to apply on checkout
operation for the current loaded patron:
  * fixed checkout end date : The fixed checkout end date is selected
    by a datepicker input field opened in a modal window.
  * override blockings : allow to ignore all circuulation blockings
    reasons (cipo, patron types restriction, patron blocked, ...)

Authored-by: Renaud Michotte <renaud.michotte@gmail.com>


## How to test

* log in no the professional view.
* go into a circulation user profile, tab 'checkin/checkout'
* Use the new setting button to set specific checkout parameters. (Note : 'override blockings' allows to override all restrictions on checkout (patron blocked, patron type restrictions, ...)
NOTE : the datepicker widget should disable all closed days for the current library. But if you are too quick, you could still choose a closed date. Don't panic, the backend will compte a correct date and you will receive a message in this case).

![image](https://user-images.githubusercontent.com/10031585/100720323-1a1f6500-33be-11eb-8f6e-c7153b7bb3a4.png)


## Code review check list
